tot/experiment.py

```diff
@@ -53,16 +53,52 @@ def __post_init__(self):
             data_params["freq"] = self.data.freq
         self.params.update({"_data_params": data_params})
         if not hasattr(self, "experiment_name") or self.experiment_name is None:
-            self.experiment_name = "{}_{}{}".format(
-                self.data.name,
-                self.model_class.model_name,
-                r"".join([r"_{0}_{1}".format(k, v) for k, v in self.params.items()])
-                .replace("'", "")
-                .replace(":", "_")
-                .replace("{", "_")
-                .replace("}", "_")
-                .replace("[", "_")
-                .replace("]", "_"),
+            # 1. Reduce and abbreviate df name (3 chars max)
+            data_name = self.data.name[:3] if len(self.data.name) > 3 else self.data.name
+            
+            # 2. Extract model name and abbreviate (first 2 chars from capitalized)
+            model_name = self.model_class.model_name
+            abbreviated_name = ""
+            for i in range(len(model_name)):
+                if model_name[i].isupper():
+                    abbreviated_name += model_name[i]+model_name[i+1]
+            abbreviated_name += "M" # for 'Model'
+            
+            # 3. params (dictionary with abbreviations)
+            d = {'seasonality_mode': 'SM',
```

Comment:
The model parameter names can largely vary among the available tsf model. Hence, it might not be a good idea of hard-coding them in a dict. Can you think of another solution?

Reply:
We should find a solution, that generalizes well to all possible model parameters